addressing/README.md

```diff
@@ -0,0 +1,337 @@
+"dnsaddr=/ip6/2604:1380:2000:7a00::1/udp/4001/quic/p2p/QmbLHAnMoJPWSCR5Zhtx6BHJX9KiKNN6tpvbUcqanj75Nb"
+"dnsaddr=/ip4/147.75.83.83/udp/4001/quic/p2p/QmbLHAnMoJPWSCR5Zhtx6BHJX9KiKNN6tpvbUcqanj75Nb"
+"dnsaddr=/dns6/ams-2.bootstrap.libp2p.io/tcp/443/wss/p2p/QmbLHAnMoJPWSCR5Zhtx6BHJX9KiKNN6tpvbUcqanj75Nb"
+```
+
+The `dnsaddr` lookup serves a similar purpose to a standard A-record DNS lookup,
+however there are differences that can be important for some use cases. The most
+significant is that the `dnsaddr` entry contains a full multiaddr, which may
+include a port number or other information that an A-record lacks, and it may
+even specify a non-IP transport. Also, there are cases in which the A-record
+already serves a useful purpose; using `dnsaddr` allows a second "namespace" for
+libp2p registrations.
+
+### TCP
+
+The libp2p TCP transport is supported in all implementations and can be used
+wherever TCP/IP sockets are accessible.
+
+Addresses for the TCP transport are of the form `<ip-multiaddr>/tcp/<tcp-port>`,
+where `<ip-multiaddr>` is a multiaddr that resolves to an IP address, as
+described in the [IP and Name Resolution section](#ip-and-name-resolution). 
+The `<tcp-port>` argument must be a 16-bit unsigned integer.
+
+### WebSockets
+
+WebSocket connections are encapsulated within TCP/IP sockets, and the WebSocket
+multiaddr format mirrors this arrangement.
+
+A libp2p WebSocket multiaddr is of the form `<tcp-multiaddr>/ws` or `<tcp-multiaddr>/wss` (TLS-encrypted), where
+`<tcp-multiaddr`> is a valid mulitaddr for the TCP transport, as [described
+above](#tcp).
+
+### QUIC
+
+QUIC sessions are encapsulated within UDP datagrams, and the libp2p QUIC
+multiaddr format mirrors this arrangement.
+
+A libp2p QUIC multiaddr is of the form `<ip-multiaddr>/udp/<udp-port>/quic`,
+where `<ip-multiaddr>` is a multiaddr that resolves to an IP address, as
+described in the [IP and Name Resolution section](#ip-and-name-resolution). 
+The `<udp-port>` argument must be a 16-bit unsigned integer in network byte order.
+
+
+### `p2p-circuit` Relay Addresses
+
+The libp2p [circuit relay protocol][relay-spec] allows a libp2p peer A to
+communicate with another peer B via a third party C. This is useful for
+circumstances where A and B would be unable to communicate directly.
+
+Once a connection to the relay is established, peers can accept incoming connections
+through the relay, using a `p2p-circuit` address.
+
+Like the `ws` WebSocket multiaddr protocol the `p2p-circuit` multiaddr does not
+carry any additional address information. Instead it is composed with two other
+multiaddrs to describe a relay circuit.
+
+A full `p2p-circuit` address that describes a relay circuit is of the form:
+`<relay-multiaddr>/p2p-circuit/<destination-multiaddr>`.
+
+`<relay-multiaddr>` is the full address for the peer relaying the traffic (the
+"relay node"), including both the transport address and the `p2p` address
+containing the relay node's peer id.
```

Comment:
```suggestion
`<relay-multiaddr>` is the full address for the peer relaying the traffic (the
"relay node").
```

I don't think we require relay addresses to contain a `p2p`, thus I suggest removing the subclause.